src/freenet/node/NewPacketFormat.java

```diff
@@ -397,8 +398,14 @@ private NPFPacket decipherFromSeqnum(byte[] buf, int offset, int length, Session
 
 		byte[] payload = Arrays.copyOfRange(buf, offset + hmacLength, offset + length);
 		byte[] hash = Arrays.copyOfRange(buf, offset, offset + hmacLength);
+		byte[] localHash = Arrays.copyOf(HMAC.macWithSHA256(sessionKey.hmacKey, payload), HMAC_LENGTH);
```

Comment:
The rest of the code, including the lines directly above this one, use `hmacLength` instead of `HMAC_LENGTH`. Please choose one.